submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,244 @@
+nexthops            : ["10.1.1.2"]
+options             : {"external_ids:{submariner"="true}"}
+priority            : 20000
+```
+
+It also programs a route in the ovn-cluster-router, to route the traffic coming from other zones destined to remote cluster IP range via the
+ovn-k8s-mp0 interface IP.
+
+```bash
+_uuid               : d55185d8-3732-45c1-ae90-4a7f8cd191f7
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.1.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+#### SubmarinerNonGWRoute Controller
+
+This controller will run as part of every route agent pod and connects to the OVN DB. When a SubmarinerNonGWRoute CR is created,
+the route-agent running on the non-GW node will update the ovn-cluster-router with a logical router policy using a priority of 20000
+to send the traffic to the remote cluster via next hop mentioned, which is the transit switch IP to the g/w node. Before adding
+the route it checks if a route exists, if so it skips adding the route again. This is required to prevent duplicate update since there
+can be more than one node in each zone and hence more than one RouteAgent.
+
+```bash
+_uuid               : 22db3005-64c5-4e32-aeb0-642423c30742
+action              : reroute
+external_ids        : {}
+match               : "ip4.dst==10.132.0.0/14"
+nexthop             : []
+nexthops            : ["169.254.0.1"]
+options             : {"external_ids:{submariner"="true}"}
+priority            : 20000
+```
+
+### Backward Compatibility
+
+With this architecture, we are removing the logical switches and routers that Submariner created in the existing implementation.
+During migration, we will need to delete these components and the routes that network-plugin-syncer installed. After that,
+the new controllers shall install the updated routes. The operator needs to be updated as well, not to create a
+network-plugin-syncer pods and remove any existing deployments.
+
+### Gateway FailOver
+
+If there are two gateway nodes , the passive one will work like a non-gateway node. It will not be responsible for creating the CRs.
+In the case of gateway fail-over all the current SubmarinerGWRoute and SubmarinerNonGWRoute will be deleted by the route agent in
+the node that is transitioning to gateway node and will be recreated with updated values.
+
+#### Open issues
+
+1. The update from older version of Submariner to a newer version will create  a datapath downtime.
+2. When multiple clusters are updated we need to check if one cluster can be done at a time. The cluster will be down
+   until all the nodes are updated.
+3. When Kubernetes cluster is updated to a version that has IC enabled, there could be a datapath downtime till the
+   Submariner g/w node is updated. Since the other nodes need the transit switch IP which will be available only when
+   the g/w node is updated.
```

Comment:
Interchange point 2 and 3 so we talk about impact on single cluster before multicluster. At end of this point mention "The cluster will be down till all nodes are updated".

For multicluster, instead of need to check, would be better if we mention the impact and suggest upgrade strategy, which could be simple as updating clusters in batches [batch can be of minimum size 1] to avoid any downtime for user applications. Way it is currently worded, it is very vague. As part of this EP we should also capture possible documentation to help users with upgrades.

Reply:
Changed the order. 

Will update the upgrade section when we complete atleast some part of the implementation and have a way to test it .
We would be more clear about the challenges then.